### PR TITLE
The speed of the new staff depends on the cheat used

### PR DESCRIPTION
**Pull request by wisegoodvin**
Staff energy and speed must determines on used cheat

### CODE REVIEW

**Comment by janisozaur** [comment]
Needs network version update

**Comment by janisozaur** [comment]
There are still CI failures

File: src/openrct2-ui/windows/Cheats.cpp [hunk 1]
```diff
@@ -1199,6 +1199,15 @@ static void window_cheats_invalidate(rct_window* w)
...
     window_cheats_misc_widgets[WIDX_WEATHER].text = WeatherTypes[gClimateCurrent.Weather];
     // Staff speed
+    _selectedStaffSpeed = gCheatCurrentStaffSpeed;
+    switch (gCheatCurrentStaffSpeed)
+    {
+        case CHEATS_STAFF_NORMAL_SPEED:
+            _selectedStaffSpeed = 1;
+            break;
+        case CHEATS_STAFF_FAST_SPEED:
+            _selectedStaffSpeed = 2;
```
Comment (janisozaur) [inline src/openrct2-ui/windows/Cheats.cpp]: Doesn't look correct, you have just assigned to this variable.

Comment (wisegoodvin) [reply src/openrct2-ui/windows/Cheats.cpp]: This is part of your source code, i'm just repeat them with needed variables: https://github.com/OpenRCT2/OpenRCT2/blob/de83a0b40b360e139e0d723df616ef8aa680ce4e/src/openrct2-ui/windows/Cheats.cpp#L761-L769

**Comment by wisegoodvin** [comment]
> 
> 
> There are still CI failures

Sorry, but CI log is not enough detailed - i does not see error is my code.

**Comment by janisozaur** [comment]
I don't know how it could be any better than the exact diff you just need to apply.

File: src/openrct2-ui/windows/Cheats.cpp (hunk 1 above)
Comment (janisozaur) [reply src/openrct2-ui/windows/Cheats.cpp]: These snippets don't look equivalent to me, but I'll check it later.

**Comment by Krutonium** [comment]
@janisozaur 

It looks like it's suggesting a change of nothing.

```
@@ -442,7 +442,7 @@
 {
     uint16_t spriteIndex;
     rct_peep* peep;
-    gCheatCurrentStaffSpeed = value; 
+    gCheatCurrentStaffSpeed = value;
 
     FOR_ALL_STAFF (spriteIndex, peep)
     {
```

**Comment by janisozaur** [comment]
@Krutonium you're not reading this correctly then.

**Comment by Krutonium** [comment]
Is it not saying "Remove `gCheatCurrentStaffSpeed = value; ` and replace it with `gCheatCurrentStaffSpeed = value; `"?

**Comment by janisozaur** [comment]
No.

**Comment by ZehMatt** [comment]
```
-....gCheatCurrentStaffSpeed.=.value;.
+....gCheatCurrentStaffSpeed.=.value;
```

**Comment by pizza2004** [comment]
Thank you @ZehMatt, I was just getting this in emails and I was just as confused as @Krutonium.

**Comment by Krutonium** [comment]
still confused

Frankly I'm still confused about where that period came from. I don't see it anywhere.

OH. It's a space.

I get it now.

**Comment by duncanspumpkin** [comment]
I don't think this will work. The information is passed on the network on load but what happens if it changes during play. I think this requires a rethink of the game command HIRE_STAFF. That game command should have a field for energy and allow it to be changed if the client has permission.

**Comment by duncanspumpkin** [comment]
I'm closing this as it needs a rethink of how it works. We recently merged a game action for hiring staff and this should probably be made into something related to that.

**Pull request closed without merging**